src/shapes/IText/ITextClickBehavior.ts

```diff
@@ -84,7 +84,7 @@ export abstract class ITextClickBehavior<
     this.__lastLastClickTime = this.__lastClickTime;
     this.__lastClickTime = this.__newClickTime;
     this.__lastPointer = newPointer;
-    this.__lastSelected = this.selected;
+    this.__lastSelected = this.selected && !this.getActiveControl();
```

Comment:
__lastSelected is at this point a wrong name.
It is used exclusively to stop the mouse up default action and does nothing else, is already private.

I would rename it to `_willEnterEditOnMouseUp' and add a JSDOC block on the top where is declared.

What do you think?

I would discourage anyone to build custom logic on those internals state flip flops

Reply:
Doesn't even need to be done now. We can do an extra PR with that.

Reply:
I agree with the name
Since it is a very crucial part of state there are use cases that you need to block the enter editing somehow.
For the majority of cases this small fix should do